### PR TITLE
make oidc-authentication filter work with toggle and values

### PR DESCRIPTION
**Pull request by tilleryd**
[sc-16967]
[sc-17391]

### CODE REVIEW

File: gm/outputs/edge.cue [hunk 1]
```diff
@@ -3,55 +3,62 @@
...
+		listener_key:                Name
+		_gm_observables_topic:       Name
+		_is_ingress:                 true
+		_enable_oidc_authentication: false
+		_oidc_endpoint:              defaults.oidc.endpoint
+		_oidc_service_url:           "https://\(defaults.oidc.domain):10808"
+		_oidc_provider:              "\(defaults.oidc.endpoint)/auth/realms/\(defaults.oidc.realm)"
+		_oidc_client_secret:         defaults.oidc.client_secret
+		_oidc_cookie_domain:         defaults.oidc.domain
+		_oidc_realm:                 defaults.oidc.realm
```
Comment (tilleryd) [inline gm/outputs/edge.cue:37]: The goal with these fields are that the operators don't need to change them. They will only come into play when `_enable_oidc_authentication` is `true`.

File: k8s/intermediates.cue [hunk 2]
```diff
@@ -4,45 +4,55 @@ import (
...
+	]
+	volumeMounts: [
+		{
+			name:      "spire-socket"
+			mountPath: "/run/spire/socket"
+		},
+		{
+			name:      "tls-certs"
+			mountPath: "/etc/proxy/tls/sidecar"
+		},
```
Comment (tilleryd) [inline k8s/intermediates.cue:35]: I will make this a toggle in a follow up PR.

File: k8s/outputs/edge.cue [hunk 3]
```diff
@@ -7,47 +7,47 @@ import (
...
+			}
+			template: {
+				metadata: {
+					labels: {"greymatter.io/cluster": Name}
+				}
+				spec: #spire_permission_requests & {
+					containers: [
+						#sidecar_container_block & {_Name: Name},
+					]
+					volumes: [] + #spire_socket_volumes + #tls_cert_volumes
```
Comment (tilleryd) [inline k8s/outputs/edge.cue:29]: I will make this a toggle in a follow up PR.

File: gm/intermediates.cue [hunk 4]
```diff
@@ -12,287 +11,310 @@ import (
...
+					{
+						any: true
+					},
+				]
+			}
+		}
+	}
+}
+
+#oidc_authentication: {
```
Comment (tilleryd) [inline gm/intermediates.cue:265]: This is the default config for oidc_authentication, there are only a subset of fields that are set by the values in `defaults.oidc` in `inputs.cue`.

**Review by jimgaspari — COMMENTED** [review]
@tilleryd You will need to make modifications to the other files in the k8s/outputs folder, or your cluster will crash

**Review by jack-bischoff — APPROVED** [review]
LGTM - confirmed cue renders as expected and was challenged by keycloak when accessing dashboard

**Review by rileyp2000 — DISMISSED** [review]

**Comment by rileyp2000** [comment]
Can also confirm cue renders and encountered keycloak redirect on the dashboard

**Comment by rileyp2000** [comment]
This is unrelated to the topic of this PR, but could we also up the commit registered for the submodule?

**Comment by jimgaspari** [comment]
Tested this branch in EKS.  Works for initial deployment without tls enabled.

**Comment by jimgaspari** [comment]
Understood, just something to think about for the future

**Comment by tilleryd** [comment]
@rileyp2000 @jack-bischoff @jimgaspari If ya'll think this is good, pleas re-approve.

**Review by jimgaspari — APPROVED** [review]
Changes have been tested and approved

**Review by rileyp2000 — APPROVED** [review]

**Pull request merged**